### DISCUSSION TITLE
[RFC] Built-in redirects and automatic redirect on slug change

### DISCUSSION THREAD

**Discussion by lenvanessen** [craftcms/cms#8856] [Ideas]
### Description
Some cms systems, like the worlds most beautiful fast and well designed Magento (pun intended), have built-in redirects.
I think would be a great addition. Whenever a slug changes, a redirect is made automatically to ensure that traffic will not be lost and organic positions will be kept.

In addition, built-in redirects would be a welcome addition as well. Extensions like Seo from Ether provide such functionality already, and we use this a lot. Especially when migrating large sites and using Apache as a sever, performance can be very slow when adding all the redirects to the .htaccess file (as it is parsed before each request). This is avoidable by using a RewriteMap, but that's a bit off topic.

**Comment by piotrpog** [comment]
@lenvanessen 
just FYI, retour plugin offers such functionality.

**Comment by nilsenpaul** [comment]
From a client's perspective, the concept of 'having to manually add a redirect or paying for a plugin to do it for me' is hard to grasp. This is something that can (and should) be handled automatically by the CMS, for every element with a public URL. When can we expect the `addRedirectOnSlugChange` config parameter 😉 ?
